### PR TITLE
Rename to `ChunkedRestResponseBodyPart`

### PR DESCRIPTION
**Pull request by DaveCTurner**
Follow-up to #104851 to rename some symbols to reflect that the class
formerly known as a `ChunkedRestResponseBody` may now only be _part_ of
the whole response body.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/rest/ChunkedRestResponseBodyPartTests.java [hunk 1]
```diff
@@ -61,20 +61,25 @@ public void testEncodesChunkedXContentCorrectly() throws IOException {
         );
 
         final List<BytesReference> refsGenerated = new ArrayList<>();
-        while (chunkedResponse.isDone() == false) {
-            refsGenerated.add(chunkedResponse.encodeChunk(randomIntBetween(2, 10), BytesRefRecycler.NON_RECYCLING_INSTANCE));
+        while (firstBodyPart.isPartComplete() == false) {
+            refsGenerated.add(firstBodyPart.encodeChunk(randomIntBetween(2, 10), BytesRefRecycler.NON_RECYCLING_INSTANCE));
         }
+        assertTrue(firstBodyPart.isLastPart());
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/rest/ChunkedRestResponseBodyPartTests.java:67]: Added this assertion

File: server/src/test/java/org/elasticsearch/rest/ChunkedRestResponseBodyPartTests.java [hunk 2]
```diff
@@ -61,20 +61,25 @@ public void testEncodesChunkedXContentCorrectly() throws IOException {
...
+            "text/plain",
+            Iterators.map(chunks.iterator(), s -> w -> w.write(s))
+        );
         final List<BytesReference> refsGenerated = new ArrayList<>();
-        while (body.isDone() == false) {
-            refsGenerated.add(body.encodeChunk(randomIntBetween(2, 10), BytesRefRecycler.NON_RECYCLING_INSTANCE));
+        while (firstBodyPart.isPartComplete() == false) {
+            refsGenerated.add(firstBodyPart.encodeChunk(randomIntBetween(2, 10), BytesRefRecycler.NON_RECYCLING_INSTANCE));
         }
+        assertTrue(firstBodyPart.isLastPart());
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/rest/ChunkedRestResponseBodyPartTests.java:82]: Added this assertion

File: server/src/test/java/org/elasticsearch/http/DefaultRestChannelTests.java [hunk 3]
```diff
@@ -797,8 +797,8 @@ public String getResponseContentTypeString() {
                 () -> channel.sendResponse(RestResponse.chunked(RestStatus.OK, firstPart, () -> {
                     assertTrue(isClosed.compareAndSet(false, true));
                     for (int i = 0; i < parts.size(); i++) {
-                        assertTrue("isDone " + i, parts.get(i).isDone());
-                        assertEquals("isEndOfResponse " + i, i == parts.size() - 1, parts.get(i).isEndOfResponse());
+                        assertTrue("isDone " + i, parts.get(i).isPartComplete());
+                        assertEquals("isEndOfResponse " + i, i == parts.size() - 1, parts.get(i).isLastPart());
```
Comment (henningandersen) [inline server/src/test/java/org/elasticsearch/http/DefaultRestChannelTests.java:801]: nit:

```suggestion
                        assertTrue("isPartComplete " + i, parts.get(i).isPartComplete());
                        assertEquals("isLastPart " + i, i == parts.size() - 1, parts.get(i).isLastPart());
```

**Review by DaveCTurner — COMMENTED** [review]
Apart from rewording some of the javadocs a little and the two added assertions (noted below) this was all mechanical.

**Review by henningandersen — APPROVED** [review]
LGTM.

**Pull request merged**